packages/charting/src/components/SankeyChart/SankeyChart.base.tsx

```diff
@@ -0,0 +1,156 @@
+import * as React from 'react';
+import { classNamesFunction, getId } from 'office-ui-fabric-react/lib/Utilities';
+import { ISankeyChartProps, ISankeyChartStyleProps, ISankeyChartStyles } from './SankeyChart.types';
+import { IProcessedStyleSet } from 'office-ui-fabric-react/lib/Styling';
+import * as d3Sankey from 'd3-sankey';
+const getClassNames = classNamesFunction<ISankeyChartStyleProps, ISankeyChartStyles>();
+
+export class SankeyChartBase extends React.Component<
+  ISankeyChartProps,
+  {
+    containerWidth: number;
+    containerHeight: number;
+  }
+> {
+  private _classNames: IProcessedStyleSet<ISankeyChartStyles>;
+  private chartContainer: HTMLDivElement;
+  private _reqID: number;
+  constructor(props: ISankeyChartProps) {
+    super(props);
+    this.state = {
+      containerHeight: 0,
+      containerWidth: 0,
+    };
+  }
+  public componentDidMount(): void {
+    this._fitParentContainer();
+  }
+
+  public componentDidUpdate(prevProps: ISankeyChartProps): void {
+    if (prevProps.shouldResize !== this.props.shouldResize) {
+      this._fitParentContainer();
+    }
+  }
+  public componentWillUnmount(): void {
+    cancelAnimationFrame(this._reqID);
+  }
+  public render(): JSX.Element {
+    const { theme, className, styles, pathColor } = this.props;
+    this._classNames = getClassNames(styles!, {
+      theme: theme!,
+      width: this.state.containerWidth,
+      height: this.state.containerHeight,
+      pathColor: pathColor,
+      className,
+    });
+    const margin = { top: 10, right: 0, bottom: 10, left: 0 };
+    const width = this.state.containerWidth - margin.left - margin.right;
+    const height =
+      this.state.containerHeight - margin.top - margin.bottom > 0
+        ? this.state.containerHeight - margin.top - margin.bottom
+        : 0;
+
+    const sankey = d3Sankey
+      .sankey()
+      .nodeWidth(5)
+      .nodePadding(6)
+      .extent([
+        [1, 1],
+        [width - 1, height - 6],
+      ]);
+
+    sankey(this.props.data.SankeyChartData!);
+    const nodeData = this._createNodes(width);
+    const linkData = this._createLinks();
+    return (
+      <div
+        className={this._classNames.root}
+        role={'presentation'}
+        // eslint-disable-next-line react/jsx-no-bind
+        ref={(rootElem: HTMLDivElement) => (this.chartContainer = rootElem)}
+      >
+        <svg width={width} height={height} id={getId('sankeyChart')}>
+          <g className={this._classNames.nodes}>{nodeData}</g>
+          <g className={this._classNames.links} strokeOpacity={0.2}>
+            {linkData}
+          </g>
+        </svg>
```

Comment:
can you make sure this works in darkmode

Reply:
We can pass colors for nodes and paths based on it will render. when we pass colors make sure it will support darkmode